lib/spack/spack/detection/_common.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+"""Common functions and data structures used by other subpackages"""
```

Comment:
I don't think this comment is necessary: it is also vague. Most of the functions here relate to updating config, so that might be useful for summarizing, although `executable_prefix` is an exception to that which suggests putting it somewhere else. In any case it's fine with me if this module stays as is but I think it would be clearer without the current module docstring so either:

* Remove it
* Update it (and possibly move some functions between modules)

Reply:
Updated in 7600804a2